packages/pwa/app/commerce-api/index.js

```diff
@@ -201,11 +202,16 @@ class CommerceAPI {
             await this.auth.login()
         }
 
+        const url = new URL(getAppOrigin())
+
         // Apply the appropriate auth headers and return new options
         const [fetchOptions, ...restParams] = params
         const newFetchOptions = {
             ...fetchOptions,
-            headers: {...fetchOptions.headers, Authorization: this.auth.authToken}
+            headers: {...fetchOptions.headers, Authorization: this.auth.authToken},
+            fetchOptions: {
+                agent: await createAgent(url.protocol)
+            }
```

Comment:
Would it be better to defined the `fetchOptions` in the `this._config`? Yo ucan see in johns [gist](https://gist.github.com/johnboxall/8d72d81de5da65a8656cdf26ee4f0e4f#file-index-mjs-L62) that he is defineding the helper as a function to switch between the protocols. But I don't see the protocol changing from call to call at that point anyway.